File: 1_GuidedExercise/idea_04.rb
class Idea04
  # Task: This function has a routine that's repeated for each word of a sentence. Extract it into its own method.
  def extract_a_method(sentence)
    words = sentence.split
    counts = []

    words.each do |word|
      current_count = 0

      word = word.downcase
      ['a', 'e', 'i', 'o', 'u'].each do |vowel|
        current_count += word.count(vowel)
      end

      counts.push(current_count)
    end

    counts
  end
end

File: js/1_GuidedExercise/src/idea_04.js
// Task: This function has a routine that's repeated for each word of a sentence. Extract it into its own method.
const extractAMethod = (sentence) => {
    const words = sentence.split(' ')
    const counts = []

    words.forEach((word) => {
        let vowellessWord = word.toLowerCase()

        const vowels = ['a', 'e', 'i', 'o', 'u']
        vowels.forEach((vowel) => {
            vowellessWord = vowellessWord.replace(new RegExp(vowel, 'g'), '')
        })

        counts.push(word.length - vowellessWord.length)
    })

    return counts
}

module.exports = {extractAMethod}
